src/YoutubeAtom.tsx

```diff
@@ -39,20 +39,28 @@ type YoutubeStateChangeEventType = { data: -1 | 0 | 1 | 2 | 3 | 5 };
 
 type EmbedConfig = {
     adsConfig: {
-        adTagParameters: {
+        adTagParameters?: {
             iu: string;
             cust_params: string;
         };
+        disableAds?: boolean;
     };
```

Comment:
I think this type should be either containing `adTagParameter` or `disableAds: true`.

Reply:
For example, in [`prebid.ts`](https://github.com/guardian/frontend/blob/5b25cc1996b4bd87a60a853e7e74820674457e0e/static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.ts#L29-L49):

```ts
type ConsentManagement =
	| {
			gdpr: GDPRConfig;
	  }
	| {
			usp: USPConfig;
	  };

type UserSync =
	| {
			syncsPerBidder: number;
			filterSettings: {
				all: {
					bidders: string;
					filter: string;
				};
			};
	  }
	| {
			syncEnabled: false;
	  };
```

Reply:
Here’s a [nice example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAogtgIwgEwMIHsB2AzAlgcygF4oBvAWACgoaoBDZAZwxwIC4yrbv7kAVOvgAKdAE504EYBFGMOFajyW4Arh0bBRuTPgDcXJTwDGKjQH0wYiXKgatO-YqUBfRwEg3yXIzoIANhAAgkwA-BzYdH6MEI7czlAAPpyUHl4+-kFMHJoqMVRurlSFlFRGWBpQAG6RuGhYePgc8Eh1rIQkCm4MzPXsyR4A9ANQab4BwTYRURAANFBDUEZ0mFBIUOhwuMDSyPld-IIi4pLSsvJ7bqocAESMG1IAFtr41zMXJuaWxzbX6MAPMjM-2erz2zny4MokNK5WAUG01T8tRYDSaiBQKIIxH63UxjX6nm8Y0yNhysz2DAEwisJxkNk6HiuUFu92BOlBKQ8H2AFhpPz+ANEQKe7LenMhBSKQA)

```ts
type EmbedConfig = {
    adsConfig: {
        adTagParameters: {
            iu: string;
            cust_params: string;
        };
		disableAds?: false;
    } | {
		disableAds: true;
	};
};

const validConfig: EmbedConfig = {
	adsConfig: {
		// disableAds: false, // can be omitted
		adTagParameters: {
			iu: "something",
			cust_params: "other_thing",
		}
	}
}

const invalidConfig: EmbedConfig = {
	adsConfig: {
		disableAds: true,
		adTagParameters: {
			iu: "something",
			cust_params: "other_thing",
		}
	}
}
```